public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -162,12 +191,33 @@ const Policy = (props: PolicyComponentProps) => {
   // policies then we should not show it. Same if the user is not supported to see autogenerated policies.
   const hideCurrentPolicy =
     isAutoGenerated && (!isAllowedToSeeAutogeneratedChunk || !isSupportedToSeeAutogeneratedChunk);
+  const hideCurrentPolicyForFilters = filtersApplied && !belongsToMatchPath;
 
-  if (hideCurrentPolicy) {
+  if (hideCurrentPolicy || hideCurrentPolicyForFilters) {
     return null;
   }
+
   const isImmutablePolicy = isDefaultPolicy || isAutogeneratedPolicyRoot;
   // TODO dead branch detection, warnings for all sort of configs that won't work or will never be activated
+
+  const childPoliciesBelongingToMatchPath = childPolicies.filter((child) =>
+    routesPath.some((route: RouteWithID) => route.id === child.id)
+  );
+
+  // child policies to render are the ones that belong to the path that matches the filters
+  const hasSomeChildPoliciesBelongingToMatchPath = childPoliciesBelongingToMatchPath.length > 0;
+  const childPoliciesToRender = !filtersApplied
+    ? childPolicies.slice(0, visibleChildPolicies) // no filters, show all
+    : hasSomeChildPoliciesBelongingToMatchPath // if we have filters, show all that match the filters
+      ? childPoliciesBelongingToMatchPath.slice(0, visibleChildPolicies)
+      : [];
```

Comment:
What is the reason for the second condition?

Is it different from:
```js
const childPoliciesToRender = !filtersApplied
    ? childPolicies.slice(0, visibleChildPolicies) // no filters, show all
    :  childPoliciesBelongingToMatchPath.slice(0, visibleChildPolicies)
```

Also, I'd consider switching the condition to be positive, so instead `!filtersApplied` just `filtersApplied`